addons/payment_stripe/models/payment_transaction.py

```diff
@@ -249,33 +277,81 @@ def _send_void_request(self):
+    def _send_refund_request(self, amount_to_refund=None, create_refund_transaction=True):
+        """ Override of payment to send a refund request to Stripe.
 
         Note: self.ensure_one()
 
-        :return: The Payment Intent
-        :rtype: dict
+        :param float amount_to_refund: The amount to refund.
+        :param bool create_refund_transaction: Whether a refund transaction should be created or
+                                               not.
+        :return: The refund transaction, if any.
+        :rtype: recordset of `payment.transaction`
         """
-        if not self.token_id.stripe_payment_method:  # Pre-SCA token -> migrate it
-            self.token_id._stripe_sca_migrate_customer()
+        if self.provider != 'stripe':
+            return super()._send_refund_request(
+                amount_to_refund=amount_to_refund,
+                create_refund_transaction=create_refund_transaction,
+            )
+        refund_tx = super()._send_refund_request(
+            amount_to_refund=amount_to_refund, create_refund_transaction=True
+        )
 
-        response = self.acquirer_id._stripe_make_request(
-            'payment_intents',
-            payload=self._stripe_prepare_payment_intent_payload(),
-            offline=self.operation == 'offline',
+        # Make the refund request to stripe.
+        data = self.acquirer_id._stripe_make_request(
+            'refunds', payload={
+                'charge': self.acquirer_reference,
+                'amount': payment_utils.to_minor_currency_units(
+                    -refund_tx.amount,  # Refund transactions' amount is negative, inverse it.
+                    refund_tx.currency_id,
+                ),
+                'metadata[reference]': refund_tx.reference,
+            }
         )
-        if 'error' not in response:
-            payment_intent = response
-        else:  # A processing error was returned in place of the payment intent
-            error_msg = response['error'].get('message')
-            self._set_error("Stripe: " + _(
-                "The communication with the API failed.\n"
-                "Stripe gave us the following info about the problem:\n'%s'", error_msg
-            ))  # Flag transaction as in error now as the intent status might have a valid value
-            payment_intent = response['error'].get('payment_intent')  # Get the PI from the error
+        _logger.info(
+            "refund request response for transaction wih reference %s:\n%s",
+            self.reference, pprint.pformat(data)
+        )
+        # Handle the refund request response.
+        notification_data = {'refund_tx': self.reference}
+        StripeController._include_refund_in_notification_data(
+            data, notification_data
+        )
+        refund_tx._handle_notification_data('stripe', notification_data)
 
-        return payment_intent
+        return refund_tx
+
+    def _stripe_get_refund_tx(self, source_tx, data):
```

Comment:
I feel like you ask to add more logic than necessary in `_get_tx_from_notification_data`.

I don't think this would be a great idea to do this in `_get_tx_from_notification_data because:
- `_get_tx_from_notification_data` is based on the reference and not on the acquirer_reference. The acquirer_reference is not always set;
- At this state in the controller, we already know that the refund was created from Stripe's dashboard and is unknown from Odoo. The check in `_stripe_get_refund_tx`is just a fail-safe in case we receive the notification more than one time to avoid duplicates.

To avoid confusion, I renamed the method to `_stripe_create_refund_tx`.